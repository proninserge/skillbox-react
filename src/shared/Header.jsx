import * as React from 'react';
import { hot } from 'react-hot-loader/root';

function HeaderComponent() {
    return (
        <header>
            <h1>Hello!</h1>
            <h2>It's me</h2>
        </header>
    );
}

export const Header = hot(HeaderComponent);