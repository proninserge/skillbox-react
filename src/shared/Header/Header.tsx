import styles from './header.less';
import { SearchBlock } from './SearchBlock/SearchBlock';
import { SortBlock } from './SortBlock/SortBlock';
import { ThreadTitle } from './ThreadTitle/TreadTitle';

export function Header() {
    return (
        <header className={styles.header}>
            <SearchBlock />
            <ThreadTitle />
            <SortBlock />
        </header>
    );
}
