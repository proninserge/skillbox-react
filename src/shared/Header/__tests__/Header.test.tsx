import {shallow, configure} from 'enzyme';
import {Header} from '../Header';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Header', () => {
    const wrapper = shallow(<Header />);

    test('should render', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.find('header.header').isEmptyRender()).toBeFalsy();
    });

    test ('should render snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});