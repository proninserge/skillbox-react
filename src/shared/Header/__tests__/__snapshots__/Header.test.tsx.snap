// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render snapshot 1`] = `
<header
  className="header"
>
  <SearchBlock />
  <ThreadTitle />
  <SortBlock />
</header>
`;
